helper/call.go

```diff
@@ -353,25 +367,45 @@ func (c *ContractCaller) IsTxConfirmed(currentTime time.Time, tx common.Hash, tx
 
 // GetConfirmedTxReceipt returns confirmed tx receipt
 func (c *ContractCaller) GetConfirmedTxReceipt(currentTime time.Time, tx common.Hash, txConfirmationTime time.Duration) (*ethTypes.Receipt, error) {
-	// get main tx receipt
-	receipt, err := c.GetMainTxReceipt(tx)
-	if err != nil {
-		Logger.Error("Error while fetching mainchain receipt", "error", err, "txHash", tx.Hex())
-		return nil, err
+	time := uint64(0)
+	var receipt *ethTypes.Receipt = nil
+
+	receiptCache, ok := c.ReceiptCache.Get(tx.String())
+
+	if !ok {
+		var err error
+
+		// get main tx receipt
+		receipt, err = c.GetMainTxReceipt(tx)
+		if err != nil {
+			Logger.Error("Error while fetching mainchain receipt", "error", err, "txHash", tx.Hex())
+			return nil, err
+		}
+
+		c.ReceiptCache.Add(tx.String(), receipt)
```

Comment:
won't it be sufficient to store only receipt.BlockNumber.Uint64() in cache?

Reply:
are we sure that, we won't require other field from Receipt in future?
then we can only store block number
That will be better

Reply:
We need receipt.